extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/extractors/data/MediaCCCRecording.java

```diff
@@ -0,0 +1,72 @@
+package org.schabi.newpipe.extractor.services.media_ccc.extractors.data;
+
+import com.grack.nanojson.JsonObject;
+
+import java.util.List;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+
+/** A recording stream of a talk/event. Switch on the implementation to get the actual data. */
+public interface MediaCCCRecording {
+
+    /** A recording stream of a talk/event.
+     * These files usually have one or more audio streams in different languages. */
+    class Video implements MediaCCCRecording {
+        public String filename;
+        public VideoType recordingType;
+        public String mimeType;
+        /** Each language is one separate audio track on the video. */
+        public List<Locale> languages;
+        public String url;
+        public int lengthSeconds;
+        public int width;
+        public int height;
+    }
+
+    /** Some talks have multiple kinds of video. */
+    enum VideoType {
+        /** The main recording of a talk/event. */
+        MAIN,
+        /** A side-recording of a talk/event that has the slides full-screen.
+         * Usually if there is a slide-recording there is a MAIN recording as well */
+        SLIDES
```

Comment:
Probably we should add something like [this](https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/stream/AudioTrackType.java) but for videos, to distinguish between the main and secondary types

Reply:
True, but it's something that should be exposed at the extractor-level APIs, not be specific for every service. This way it would be easy in clients to add a switch that allows choosing the video track type.

Reply:
Yes, maybe, once we figured out which parts are specific to media.ccc.de and which ones can be abstracted :)

I think a major issue with NPE is that it’s trying to unify the services too much, and then we get lots of workarounds on the NewPipe side cause we have to reconstruct specific service behaviour again that was unified on the NPE side. I’d rather cast on the class and then have some service-specific functionality available if I need it in the player.